### PR TITLE
Added filter for nonva meds

### PR DESCRIPTION
**Pull request by agravell047**
## Summary

Added logic to remove all Non-VA medications that don't have a display status of 'Active: Non-VA'.

Also cleaned up two controller methods.

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-52630
>User had some medications in "Discontinued" status, but the record had no Medication Name so there was no way to click to see more details.
 
>For Non-VA medications: * [va.gov](http://va.gov/) will use the "prescriptionSource" field to determine if a medication is >Non-VA by the presence of "NV" in that field

>[va.gov](http://va.gov/) API will only send NVs with a status of Active: Non-VA Medication
>NVs in other statuses will NOT display or send via API to the Veteran

## Testing done

- Locally tested
- Ran spec tests locally

## Screenshots
No UI Changes

## What areas of the site does it impact?
MHV Medications List page

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by vmenshutin-bylight — APPROVED** [review]

**Review by vmenshutin-bylight — APPROVED** [review]

File: modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb [hunk 1]
```diff
@@ -56,6 +57,10 @@ def collection_resource
         end
       end
 
+      def filter_non_va_meds(data)
+        data.reject { |item| item[:prescription_source] == 'NV' && item[:disp_status] != 'Active: Non-VA' }
```
Comment (robertbylight) [inline modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb:61]: Would you be able to cover this logic in the spec?

**Review by robertbylight — CHANGES_REQUESTED** [review]

File: modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb (hunk 1 above)
Comment (agravell047) [reply modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb:61]: @robertbylight I tried covering this in a spec but the data returned from the API doesn't have any 'invalid' data like we saw on the Veteran's account. Is there a way you can think of that I might be missing?

Comment (robertbylight) [reply modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb:61]: You could try to copy a cassette rename it and hardcode the invalid data.

**Review by robertbylight — APPROVED** [review]

File: modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb (hunk 1 above)
Comment (agravell047) [reply modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb:61]: also discussed adding 'invalid' data to our test data returned from the API and he said that wouldn't be an option. So I don't think there is a way to create a valid test for this scenario

**Review by robertbylight — APPROVED** [review]

**Review by vmenshutin-bylight — APPROVED** [review]

**Pull request merged**